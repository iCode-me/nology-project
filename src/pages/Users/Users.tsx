import { Fragment, useCallback, useEffect, useState } from "react";
import Layout from "../../components/Layout/Layout";
import { getUsers } from "../../data/api";
import UserData from "../../types/internal/UserData";

// to see this page component action, append this to your URL: "/users" 🧐
// TODO: Add section in readme breaking down this component
const Users = () => {
  const [users, setUsers] = useState<UserData[] | null>();

  useEffect(() => {
    (async () => {
      setUsers((await getUsers()) ?? null);
    })();
  }, []);

  const usersTsx = useCallback(
    (users: UserData[] | null) =>
      users && users.length > 0 ? (
        users.map((user) => (
          <Fragment key={user.id}>
            <h2>{user.fullName}</h2>
            <p>User Id: {user.id}</p>
          </Fragment>
        ))
      ) : (
        <span>Sorry, no users found</span>
      ),
    []
  );

  return (
    <Layout>
      <h1>Users</h1>
      {users !== undefined ? usersTsx(users) : <span>Loading...</span>}
    </Layout>
  );
};

export default Users;
