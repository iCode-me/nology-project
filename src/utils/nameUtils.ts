export const formatName = (firstName: string, lastName: string): string => {
  return `${firstName} ${lastName}`;
};

// export const formatNameArray = (names: string[]): string => {
//   return names.join(" ");
// };
