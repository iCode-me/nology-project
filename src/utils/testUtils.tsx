import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

type CustomRenderOptions = {
  useRouting: boolean;
};

export const customRender = (
  ui: JSX.Element,
  options: CustomRenderOptions = {
    useRouting: true,
  }
) => {
  let uiResult = ui;

  if (options.useRouting) {
    uiResult = wrapWithRouting(ui);
  }

  return render(uiResult);
};

const wrapWithRouting = (ui: JSX.Element): JSX.Element => {
  return <BrowserRouter>{ui}</BrowserRouter>;
};
