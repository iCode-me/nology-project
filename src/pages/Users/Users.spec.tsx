import { customRender } from "../../utils/testUtils";
import Users from "./Users";

export default describe("Users Page", () => {
  it("Renders component in loading state", () => {
    const { container } = customRender(<Users />);
    expect(container).toMatchSnapshot();
  });

  //TODO: Add mock data to show component with data
});
