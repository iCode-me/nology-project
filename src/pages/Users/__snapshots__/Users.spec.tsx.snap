// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Page Renders component in loading state 1`] = `
<div>
  <div
    class="layout"
  >
    <main
      class="layout__main"
    >
      <h1>
        Users
      </h1>
      <span>
        Loading...
      </span>
    </main>
  </div>
</div>
`;
